linked-lists/c-or-cpp/sortedInsert.cpp

```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <time.h>
+
+template <typename T>
+class LinkedList{
+	public:
+		void push(const T& data);
+		void display();
+		void sortedInsert(const bool& reverse = false);
+		void pushfront(const T& data);
+
+	private:
+		bool reverse_;
+		bool sortInsertInit_ = false;
+		void sortedInsertALGO();
+		struct node{
+			T data;
+			node* next;
+			node(const T& data): data(data), next(nullptr){}
+		};
+		node* head_ = nullptr;
+};
+
+template <typename T>
+void LinkedList<T>::sortedInsert(const bool& reverse){
+	/*
+	Sorts the list in either order as it inserts new element to the list
+
+	@params: reverse
+		false(default) - for Ascending order
+		true - for Descending order
+	*/
+	
+	if(this->head_ != nullptr){
+		std::cout << "Can only be called before pushing any element to the list" << std::endl;	
+		return;
+	}
+
+	reverse_ = reverse;
+	sortInsertInit_ = true;
+}
+
+
+template <typename T>
+void LinkedList<T>::pushfront(const T& data){
+	node* n = new node(data);
+	n->next = head_;
+	head_ = n;
+}
+
+template <typename T>
+void LinkedList<T>::push(const T& data){
+	if(!sortInsertInit_){
+		std::cout << "Initialize sortInsert before calling push" << std::endl;
```

Comment:
If triggering sortedInsert() in a mandatory for push() to work then what is the reason for it to be an additional function ? 

you could have passed it as a boolean value as an argument to the push() function only. Also, if boolean value represents the list being ascending or descending then condition `if(!sortInsertInit_)` will always be true for descending list also.

---

Prefer using stderr ( instead of stdout ) to display error messages. or throw exceptions in such cases.

Reply:
would it be better if i do something like this?
`l1.sortInsertRev = true`
 this is for descending order, so by default it will be in ascending order.

So if I want to it to be that way it has to be set to `true` after creating the list before pushing anything.
If i do it after i push some integers the list order will be messed up, some integers will be in ascending and some will be in descending.
Should i avoid such events and write an if statement or leave it like that?